src/main/java/net/sf/jabref/specialfields/SpecialField.java

```diff
@@ -58,11 +57,35 @@ public SpecialFieldValue parse(String s) {
 
     public abstract String getFieldName();
 
+    public abstract String getLocalizedFieldName();
+
     public abstract Icon getRepresentingIcon();
 
-    public abstract String getMenuString();
+    public String getMenuString() {
+        return getLocalizedFieldName();
+    }
+
+    public String getToolTip() {
+        return getLocalizedFieldName();
+    }
+
+    public String getTextDone(String... params) {
+        if(isSingleValueField()) {
+            if(params.length == 1 && params[0] != null) {
+                return Localization.lang("Toggled '%0' for %1 entries", getLocalizedFieldName(), params[0]);
+            }
+        } else {
+            if (params.length == 2 && params[0] != null) {
+                String[] allParams = {getLocalizedFieldName(), params[0], params[1]};
+                return Localization.lang("Set '%0' to '%1' for %2 entries", allParams);
+            } else if (params.length == 1) {
+                return Localization.lang("Cleared '%0' for %1 entries", getLocalizedFieldName(), params[0]);
+            }
+        }
```

Comment:
Hard to read. Can you make it more ovious which three results are to be returned? Maybe only one level of indent would help here. Also what happens if I pass in null only and it is not a single value field?